README.Rmd

```diff
@@ -0,0 +1,116 @@
+---
+output: github_document
+---
+
+<!-- README.md is generated from README.Rmd. Please modify README.Rmd and run `pkgdown::build_site()` to update README.md -->
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.path = "README-"
+)
+```
+
+# synapserutils
+
+The `synapserutils` package provides a set of utility functions, built on top of [`synapser`](http://sage-bionetworks.github.io/synapser/) package.
+
+## Installation
+
+`synapserutils` is available as a ready-built package for Microsoft Windows and Mac OSX. For Linux systems, it is available to install from source. It can be installed or upgraded using the standard `install.packages()` command, adding the [Sage Bionetworks R Archive Network (RAN)](https://sage-bionetworks.github.io/ran) to the repository list, e.g.:
+
+```{r eval=F}
+install.packages("synapserutils", repos=c("https://sage-bionetworks.github.io/ran", "http://cran.fhcrc.org"))
+```
+Alternatively, edit your `~/.Rprofile` and configure your default repositories:
+```{r eval=F}
+options(repos=c("https://sage-bionetworks.github.io/ran", "http://cran.fhcrc.org"))
+```
+after which you may run `install.packages` without specifying the repositories:
+```{r eval=F}
+install.packages("synapserutils")
+```
+
+If you have been asked to validate a release candidate, please replace the URL https://sage-bionetworks.github.io/ran with https://sage-bionetworks.github.io/staging-ran, that is:
+
+```{r eval=F}
+install.packages("synapserutils", repos=c("https://sage-bionetworks.github.io/staging-ran", "http://cran.fhcrc.org"))
+```
+
+## Usage
+
+To get started, try logging into Synapse. If you don't already have a Synapse account, register [here](https://www.synapse.org/register):
+
+```{r eval=F}
+library(synapserutils)
+synLogin()
+```
+
+## Available Utilities
+
+### Copy
+
+#### Copy Entity (File, Folder, Table, Link, and Project)
+
+The example below copy File ID `syn123` to Project ID `syn456`:
+```{r eval=F}
+  copy("syn123", "syn456")
+```
+
+The first parameter in `copy()` can be a File ID, a Table ID, or a Link ID. The second parameter must be a Folder ID or a Project ID.
```

Comment:
It appears that copying a folder also works. The documentation for copy() doesn't state what types of entities can be copied; can we put this information there as well?

Reply:
For this topic, the `copy` function in the Python client is not tested. I want to hold off on advertising copying folder (and possibly recursively copy the entire hierarchy) until we refactor and have all test coverage for this function. This refactor would also include the re-write/ edit the `copy` function documentation in Python. synapserutils will then inherit the new docs. I will update this part then.